test.sh

```diff
@@ -64,10 +97,107 @@ function run_test_coverage() {
   goveralls -v -coverprofile=gover.coverprofile -service=travis-pro
 }
 
+#
+# Main CLI Parser
+#
+USAGE="$(cat -- <<-EOM
+
+Usage:
+Boulder test suite CLI, intended to be run inside of a Docker container:
+
+  docker-compose run --use-aliases boulder ./$(basename "${0}") [OPTION]...
+
+With no options passed: runs standard battery of tests (lint, unit, and integation)
+    -l, --lints                           Adds lint to the list of tests to run
+    -u, --unit                            Adds unit to the list of tests to run
+    -d, --unit-test-filter <DIRECTORY>    Run unit tests for a specific directory
+    -e, --enable-unit-race                Enables -race flag for all unit tests
+    -n, --config-next                     Changes BOULDER_CONFIG_DIR from test/config to test/config-next
+    -c, --coverage                        Adds coverage to the list of tests to run
+    -i, --integration                     Adds integration to the list of tests to run
+    -s, --start-py                        Adds start (py) to the list of tests to run
+    -v, --gomod-vendor                    Adds gomod-vendor to the list of tests to run
+    -g, --generate                        Adds generate to the list of tests to run
+    -r, --rpm                             Adds rpm to the list of tests to run
+    -p, --show-integration-test-list      Outputs a list of the available integration tests
+    -f, --integration-filter <REGEX>      Run only those tests and examples matching the regular expression
+
+                                          Note:
+                                           This option disables the '"back in time"' integration test setup
+
+                                           For tests, the regular expression is split by unbracketed slash (/)
+                                           characters into a sequence of regular expressions
+
+                                          Example:
+                                           TestAkamaiPurgerDrainQueueFails/TestWFECORS
+    -h, --help                            Shows this help message
+
+EOM
+)"
+
+while getopts luecipsvgrnhd:f:-: OPT; do
+  if [ "$OPT" = - ]; then   # long option: reformulate OPT and OPTARG
```

Comment:
```suggestion
  if [ "$OPT" = - ]; then     # long option: reformulate OPT and OPTARG
```

(Added to spaces to line up the # with subsequent lines)